syntaxes/happy.YAML-tmLanguage

```diff
@@ -0,0 +1,75 @@
+fileTypes:
+      - y
+name: Haskell Happy
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  comments:
+    patterns:
+      - name: comment.line
+        match: \-\-.*$
+      - name: comment.block
+        begin: "{-"
+        end: "-}"
+  blocks:
+    name: meta.embedded.block.haskell
+    begin: "{"
+    beginCaptures:
+      "0": { name: punctuation.block.begin.happy }
+    end: "}"
+    endCaptures:
+      "0": { name: punctuation.block.end.happy }
+    patterns:
+      - include: source.haskell
```

Comment:
Blocks can also be monadic in Happy, which means they start with `{%` instead of just `{`.